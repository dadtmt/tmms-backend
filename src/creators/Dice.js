import React, { Component, PropTypes } from 'react'
import { Checkbox, Panel } from 'react-bootstrap'

import FieldGroup from '../components/FieldGroup'

class Dice extends Component {
  constructor(props) {
    super(props)
    this.state = {
      master: false,
      modifier: 0,
      nbDices: 1,
      nbSides: 6
    }
    this.handleChange = this.handleChange.bind(this)
    this.handleNbDicesChange = this.handleNbDicesChange.bind(this)
    this.handleNbSidesChange = this.handleNbSidesChange.bind(this)
    this.handleModifierChange = this.handleModifierChange.bind(this)
    this.handleMasterChange = this.handleMasterChange.bind(this)
  }

  componentDidMount() {
    this.handleChange()
  }

  handleChange() {
    this.props.onChange(this.state)
  }

  handleNbDicesChange(event) {
    this.setState({ nbDices: event.target.value })
    this.handleChange()
  }

  handleNbSidesChange(event) {
    this.setState({ nbSides: event.target.value })
    this.handleChange()
  }

  handleModifierChange(event) {
    this.setState({ modifier: event.target.value })
    this.handleChange()
  }

  handleMasterChange() {
    this.setState({ master: !this.state.master })
    this.handleChange()
  }

  render() {

    return (
      <Panel
        header='Add a Test'
      >
        <Checkbox
          checked={this.state.master}
          onClick={this.handleMasterChange}
        >
          The story teller rolls the dice!
        </Checkbox>
        <FieldGroup
          id='nbDices'
          type='number'
          label='Number of dices'
          onChange={this.handleNbDicesChange}
          value={this.state.nbDices}
        />
        <FieldGroup
          id='nbSides'
          type='number'
          label='Number of sides'
          onChange={this.handleNbSidesChange}
          value={this.state.nbSides}
        />
        <FieldGroup
          id='modifier'
          type='number'
          label='modifier'
          onChange={this.handleModifierChange}
          value={this.state.modifier}
        />
      </Panel>
    )
  }
}

Dice.propTypes = {
  onChange: PropTypes.func.isRequired
}

export default Dice
