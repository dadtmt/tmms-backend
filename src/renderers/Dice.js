import React, { PropTypes } from 'react'
import { Label } from 'react-bootstrap'

const Dice = ({
  details,
  made,
  master,
  modifier,
  nbDices,
  nbSides,
  result
}) =>
    <div>
      <Label>Lancer les dés: {`${nbDices}D${nbSides}+${modifier}`}</Label>
        {
          (made || master) &&
            <div>
              <Label>Résultat: {result}</Label>
              <Label>Détails: {details}</Label>
            </div>
        }
    </div>

Dice.propTypes = {
  details: PropTypes.string,
  made: PropTypes.bool.isRequired,
  master: PropTypes.bool.isRequired,
  modifier: PropTypes.number.isRequired,
  nbDices: PropTypes.number.isRequired,
  nbSides: PropTypes.number.isRequired,
  result: PropTypes.number
}

export default Dice
