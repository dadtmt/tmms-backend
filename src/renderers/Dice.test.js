import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import Dice from './Dice'

const notResolvedprops = {
  made: false,
  master: false,
  modifier: 0,
  nbDices: 1,
  nbSides: 6
}

const props = {
  details: 'details',
  made: false,
  master: false,
  modifier: 0,
  nbDices: 1,
  nbSides: 6,
  result: 6
}

it('renders without crashing', () => {
  const div = document.createElement('div')

  ReactDOM.render(
    <Dice {...props} />,
    div
  )

  ReactDOM.render(
    <Dice {...notResolvedprops} />,
    div
  )
})

it('renders hiding details and result when master and made false', () => {
  expect(
    renderer.create(<Dice {...props} />).toJSON()
  ).toMatchSnapshot()
})

it('renders showing details and result when master or made true', () => {
  expect(
    renderer.create(<Dice {...props} master={true} />).toJSON()
  ).toMatchSnapshot()
  expect(
    renderer.create(<Dice {...props} made={true} />).toJSON()
  ).toMatchSnapshot()
  expect(
    renderer.create(<Dice {...props} master={true} made={true} />).toJSON()
  ).toMatchSnapshot()
})
