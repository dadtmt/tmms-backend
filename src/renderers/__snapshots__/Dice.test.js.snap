exports[`test renders hiding details and result when master and made false 1`] = `
<div>
  <span
    className="label label-default">
    Lancer les dés: 
    1D6+0
  </span>
</div>
`;

exports[`test renders showing details and result when master or made true 1`] = `
<div>
  <span
    className="label label-default">
    Lancer les dés: 
    1D6+0
  </span>
  <div>
    <span
      className="label label-default">
      Résultat: 
      6
    </span>
    <span
      className="label label-default">
      Détails: 
      details
    </span>
  </div>
</div>
`;

exports[`test renders showing details and result when master or made true 2`] = `
<div>
  <span
    className="label label-default">
    Lancer les dés: 
    1D6+0
  </span>
  <div>
    <span
      className="label label-default">
      Résultat: 
      6
    </span>
    <span
      className="label label-default">
      Détails: 
      details
    </span>
  </div>
</div>
`;

exports[`test renders showing details and result when master or made true 3`] = `
<div>
  <span
    className="label label-default">
    Lancer les dés: 
    1D6+0
  </span>
  <div>
    <span
      className="label label-default">
      Résultat: 
      6
    </span>
    <span
      className="label label-default">
      Détails: 
      details
    </span>
  </div>
</div>
`;
